include/openthread/netdiag.h

```diff
@@ -128,6 +129,46 @@ typedef struct otNetworkDiagRoute
     otNetworkDiagRouteData mRouteData[OT_NETWORK_MAX_ROUTER_ID + 1]; ///< Link Quality and Routing Cost data.
 } otNetworkDiagRoute;
 
+/**
+ * Represents a Network Diagnostic Enhanced Route data.
+ */
+typedef struct otNetworkDiagEnhRouteData
+{
+    uint8_t mRouterId;           ///< The Router ID.
+    bool    mIsSelf : 1;         ///< This is the queried device itself. If set, the other fields should be ignored.
+    bool    mHasLink : 1;        ///< Indicates whether the queried device has a direct link with router.
+    uint8_t mLinkQualityOut : 2; ///< Link Quality Out (applicable when `mHasLink`).
+    uint8_t mLinkQualityIn : 2;  ///< Link Quality In (applicable when `mHasLink`).
+
+    /**
+     * The next hop Router ID tracked towards this router.
+     *
+     * If the device has no direct link with the router (`mHasLink == false`), this field indicates the next hop router
+     * that would be used to forward messages destined to that router.
+     *
+     * If the device has a direct link with the router (`mHasLink == true`), this field indicates the backup next hop
+     * that would be used if the direct link is no longer usable.
+     *
+     * If there is no next hop, then `OT_NETWORK_MAX_ROUTER_ID + 1` will be used.
+     */
+    uint8_t mNextHop;
+
+    /**
+     * The cost associated with forwarding using Next Hop (regardless of whether the device has a direct ink to
```

Comment:
```suggestion
     * The cost associated with forwarding using Next Hop (regardless of whether the device has a direct link to
```

Reply:
Is this the reported cost R_route_cost per Section 5.9.2 ?
This would be the best I think. In that case, the actual route cost to go from queried-Router to Router(ID) is equal to:

R_route_cost + Cost(direct link from queried-Router to Router(NextHop) )

The latter Cost() function can be determined from LQIn and LQOut using the calculation in Section 5.9.4.

Doing it this way allows the field NHCost for example to contain 15, and LQIn and LQOut both LQ1 (lowest), then the Cost(direct link) = 4 and total route cost is 15+4 = 19.

Although this would be above max-cost, it's useful as a debugging tool to know that the route cost from the queried router would go above max-cost, and why that is the case.

Reply:
Yes. Exactly. 

I changed to this:
```c++
    /**
     * The route cost associated with forwarding to `mRouterId` using `mNextHop` (when valid).
     *
     * This is the route cost `mNextHop` has claimed to have towards `mRouterId`. Importantly, it does not include the
     * link cost to send to `mNextHop` itself.
     */
    uint8_t mNextHopCost;
```

Reply:
Ok, that's clear now I think.